### PR TITLE
Added @Toggle annotation

### PR DESCRIPTION
**Pull request by ricbra**
Inspired by Symfony's <code>SecurityListener</code> I've created a similar solution for a Toggle annotation.

When the toggle isn't active a 404 exception is thrown.

### CODE REVIEW

**Comment by wjzijderveld** [comment]
Nice, thanks for this addition!

I want to take another look at it (don't have that much time at the moment), but at first glance it looks good!

File: Tests/EventListener/ToggleListenerTest.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+
+class ToggleListenerTest extends \PHPUnit_Framework_TestCase
+{
+    public function setUp()
+    {
+        $this->request = $this->createRequest();
+
+        // trigger the autoloading of the @Toggle annotation
+        class_exists('Qandidate\Bundle\ToggleBundle\Annotations\Toggle');
```
Comment (wjzijderveld) [inline Tests/EventListener/ToggleListenerTest.php]: I would prefer that we replace this with something like this in `test/boostrap.php`:
```php
use Doctrine\Common\Annotations\AnnotationRegistry;

$loader = require __DIR__ . '/../vendor/autoload.php';

AnnotationRegistry::registerLoader(array($loader, 'loadClass'));
```

**Comment by wjzijderveld** [comment]
Minor comment, but otherwise :+1:

**Comment by ricbra** [comment]
That was indeed a bit hacky. Learned how to autoload the annotations, thanks for that.

**Comment by Brainshack** [comment]
Hi,

I personally would prefer if it were called either @Feature or @FeatureToggle. IMHO, that makes it much clearer, what this annotation does.

**Comment by wjzijderveld** [comment]
Thanks @ricbra 

@Brainshack I can understand that, but on the other hand is it still a toggle, not necessarily a feature.

**Pull request merged**